File: main/src/main/java/com/sedmelluq/discord/lavaplayer/natives/ConnectorNativeLibLoader.java
package com.sedmelluq.discord.lavaplayer.natives;

import com.sedmelluq.lava.common.natives.NativeLibraryLoader;
import com.sedmelluq.lava.common.natives.architecture.DefaultOperatingSystemTypes;

/**
 * Methods for loading the connector library.
 */
public class ConnectorNativeLibLoader {
    private static final NativeLibraryLoader[] loaders = new NativeLibraryLoader[]{
        NativeLibraryLoader.createFiltered(ConnectorNativeLibLoader.class, "libmpg123-0",
            it -> it.osType == DefaultOperatingSystemTypes.WINDOWS),
        NativeLibraryLoader.create(ConnectorNativeLibLoader.class, "connector")
    };

    /**
     * Loads the connector library with its dependencies for the current system
     */
    public static void loadConnectorLibrary() {
        for (NativeLibraryLoader loader : loaders) {
            loader.load();
        }
    }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/natives/ConnectorNativeLibLoader.kt
package com.sedmelluq.discord.lavaplayer.natives

import com.sedmelluq.lava.common.natives.NativeLibraryLoader
import com.sedmelluq.lava.common.natives.architecture.DefaultOperatingSystemTypes

/**
 * Methods for loading the connector library.
 */
object ConnectorNativeLibLoader {
    private val loaders = listOf(
        NativeLibraryLoader.createFiltered(ConnectorNativeLibLoader::class.java, "libmpg123-0") {
            it.osType === DefaultOperatingSystemTypes.WINDOWS
        },
        NativeLibraryLoader.create(ConnectorNativeLibLoader::class.java, "connector")
    )

    /**
     * Loads the connector library with its dependencies for the current system
     */
    fun loadConnectorLibrary() {
        for (loader in loaders) {
            loader.load()
        }
    }
}
